.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CI
+
+on:
+  push:
+    branches: [ main]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+  APPNAME : test-flip-link-app
+  CORE_TARGET: thumbv7em-none-eabi # needed by `core`
+  NRF_TARGET: thumbv7em-none-eabihf
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        rust:
+          - stable
+          #- nightly
+        os:
+          - ubuntu-latest
+          # FIXME: Reenable macOS, Windows, and nightly once demt-test is open-sourced.
+          #- macOS-latest
+          #- windows-latest
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: ${{ matrix.rust }}
+        override: true
+    - name: Build
+      run: RUSTFLAGS='--deny warnings' cargo build
+      shell: bash
+    - name: Check fmt
+      if: matrix.os == 'ubuntu-latest' # we only need to check fmt on one OS
+      run: cargo fmt --all -- --check
+
+  build_app:
+    strategy:
+      matrix:
+        rust:
+          - stable
+          #- nightly
+        os:
+          - ubuntu-latest # doesn't work on macOS because `cortex-m-rt` doesn't build there
```

Comment:
(I think this comment doesn't apply for this repo because we don't have a single workspace)